scipy/optimize/func.py

```diff
@@ -0,0 +1,163 @@
+"""
+Implementations of fitting algorithms for common functions that can be
+computed analytically.
+
+Functions
+---------
+.. autosummary::
+   :toctree: generated/
+
+    exp_fit
+    pow_fit
+"""
+
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['exp_fit', 'pow_fit']
+
+import numpy as np
+from numpy import (
+    argsort, asfarray, cumsum, diff, empty, empty_like, exp, log,
+    square,
+)
+from numpy.linalg import inv
+
+
+def exp_fit(x, y, sorted=True):
+    """
+    Fit an exponential curve to raveled 1D data.
+
+    This algorithm does not require any a-priori knowledge of the data,
+    such as the intercept. The fitting parameters are comptued for:
+
+    .. math::
+
+       y = A + Be^{Cx}
+
+    Parameters
+    ----------
+    x : array-like
+        The x-values of the data points. The fit will be performed on a
+        raveled version of this array.
+    y : array-like
+        The y-values of the data points corresponding to `x`. Must be
+        the same size as `x`. The fit will be performed on a raveled
+        version of this array.
+    sorted : bool
+        Set to True if `x` is already monotonically increasing or
+        decreasing. If False, x will be sorted into increasing order,
+        and y will be sorted along with it.
+
+    Return
+    ------
+    a, b, c : array
+        A 3-element array of optimized fitting parameters. The first
+        element is the additive bias, the second the multiplicative, and
+        the third the exponential.
+
+    Notes
+    -----
+    The fit is computed non-iteratively in a single pass. It can be used
+    to initialize other regression methods based on different
+    optimization criteria. The algorithm and the theory behind it is
+    presented in the paper below.
+
+    References
+    ----------
+    Jacquelin, Jean. “REGRESSIONS Et EQUATIONS INTEGRALES” 14 Jan. 2009, pp. 15–18., https://www.scribd.com/doc/14674814/Regressions-et-equations-integrales
+    """
```

Comment:
Spaces are not needed necessarily (but I don't have an opinion) but the line is too long and needs linebreaks. 

A personal note: I don't think we need to follow the school of surname, name in the documentation here. The link points to the newer version hence the date is ...ehm... outdated. Also the double quotes can be straightened out.

Reply:
@ilayn I'd be happy to insert a newline, remove the date, fix the quotes and put in line breaks. Would inserting a line break cause any problem in the output formatting? Do I need to do anything besides indent the second line by four spaces?

Reply:
No it will be just like Notes section and that would be enough. Here is the current rendering https://8282-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.optimize.exp_fit.html#scipy.optimize.exp_fit

This is kind of what other function references are doing 

```python
def exp_fit(x, y, sorted=True):
    """
    [....]

    References
    ----------
    Jean Jacquelin, "Régressions et Equations Intégrales" pp. 15–18.,
    Available Online :
    https://www.scribd.com/doc/14674814/Regressions-et-equations-integrales

    """
```

However I'm still skeptical about adding this into the references (as Warren mentioned) since there is nowhere else we can verify its integrity or content. This doesn't mean that the work is bad or wrong but we have no way to back it up.

Reply:
@madphysicist  For reference: https://www.numpy.org/devdocs/docs/howto_document.html#sections